include/mbedtls/mbedtls_config.h

```diff
@@ -3616,6 +3617,8 @@
 //#define MBEDTLS_PLATFORM_VSNPRINTF_MACRO    vsnprintf /**< Default vsnprintf macro to use, can be undefined */
 //#define MBEDTLS_PLATFORM_NV_SEED_READ_MACRO   mbedtls_platform_std_nv_seed_read /**< Default nv_seed_read function to use, can be undefined */
 //#define MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO  mbedtls_platform_std_nv_seed_write /**< Default nv_seed_write function to use, can be undefined */
+//#define MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO   int64_t /**< Default millionseconds time macro to use, can be undefined. MBEDTLS_HAVE_TIME must be enabled. It MUST be signed 64bit integer at least */
```

Comment:
millionseconds? Should be milliseconds? Unclear from the comment if it must be signed (would unsigned 128 bits be OK?)
```suggestion
//#define MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO   int64_t /**< Default milliseconds time macro to use, can be undefined. MBEDTLS_HAVE_TIME must be enabled. It may or may not be signed, and must be at least 64 bits */
```

Reply:
Err, it should be signed, as it's a time difference (like `time_t`)

Reply:
It can be 30 bits signed integer.  

It is used for time difference. And in TLS 1.3 , the max allowed difference is 7*24*3600*1000 milliseconds.  A 30 bits signed integer is enough.

Reply:
Err, sorry, I don't understand. So can it be `int32_t`? Because I thought the comment originally said it had to be at least 64 bits?

Should we now have `It must be signed, and at least 32 bits. If it is changed from the default, MBEDTLS_PRINTF_MS_TIME must be updated to match.`?

I don't think there is such a thing as a 30-bit integer in standard C: you'll pretty much only get 32 or 64 bits (not even 48!).

Reply:
Also, if we only need 30 bits, why are we defaulting it to 64 bits, not 32?

Reply:
(side note: just came across: Harris H800 had 48-bit `long` and Unisys 1100 had 36 bit `int`, but they are definitely the exceptions!)

Reply:
Sorry for that, I made a mistake. 30bits integer can work, but has some risk. I will revert it.

https://datatracker.ietf.org/doc/html/rfc8446#section-4.6.1, `ticket_lifetime` max value is 7 days.  So the time can be minimal to 7days ( 30bit signed integer). 
But that will cause possible wrong if the device runs longger than 7 days(30bits) and 25 days(32bits).

>I don't think there is such a thing as a 30-bit integer in standard C: you'll pretty much only get 32 or 64 bits (not even 48!).
It is not in standard C. but some embed device has that. I think it is over design.